src/ui/public/visualize/loader/embedded_visualize_handler.ts

```diff
@@ -250,45 +255,42 @@ export class EmbeddedVisualizeHandler {
 
   /**
    * renders visualization with provided data
-   * @param visData: visualization data
+   * @param resp: visualization data
    */
-  public render = (visData: any = null) => {
-    return visualizationLoader
-      .render(this.element, this.vis, visData, this.uiState, {
+  public render = (resp: any = null) => {
```

Comment:
we should probably not use renderer here ... and then the if condition for enable_pipeline_data_loader should come earlier ....

or, if we can actually use the renderer, maybe lets modify the output from visualize_data_loader so that it will return something like 
```
{
          as: 'visualization',
          value: {
            visType: 'histogram',
            visData: resp,
            visConfig: params,
            params: {},
          },
```

that would then also mean we don't need to handle special case in discover ...